templates/ingress.yaml

```diff
@@ -1,72 +1,55 @@
-{{ if $.Values.haproxy.ingressEnable -}}
-{{ if eq $.Values.haproxy.ingress.class "alb" -}}
-apiVersion: networking.k8s.io/v1beta1
-kind: Ingress
-metadata:
-  name: ingress
-  namespace: {{ $.Values.namespace }}
-  annotations:
-    alb.ingress.kubernetes.io/group.name: {{ $.Values.haproxy.ingress.albGroup }}
-    kubernetes.io/ingress.class: alb
-    alb.ingress.kubernetes.io/scheme: internet-facing
-    alb.ingress.kubernetes.io/target-type: ip
-    {{ if $.Values.haproxy.ingress.tlsEnable }}
-    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS":443}]'
-    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}'
-    {{ else }}
-    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}]'
-    {{ end }}
-
-  labels:
-    scope: jitsi
-
-spec:
-  rules:
-  - host: {{ $.Values.haproxy.ingress.host }}
-    http:
-      paths:
-      {{ if $.Values.haproxy.ingress.tlsEnable -}}
-      - path: /*
-        backend:
-          serviceName: ssl-redirect
-          servicePort: use-annotation
-      {{ end -}}
-      - path: /*
-        backend:
-          serviceName: haproxy
-          servicePort: 8080
-{{ else -}}
+{{ if .Values.ingress.enabled -}}
+{{ $hosts := (gt (len $.Values.ingress.hosts) 0) | ternary ($.Values.ingress.hosts) (list ($.Values.PUBLIC_URL | default "" | replace "https://" "")) -}}
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   annotations:
-    {{- if $.Values.haproxy.certmanager.issuer }}
-    cert-manager.io/cluster-issuer: {{ $.Values.haproxy.certmanager.issuer }}
+    {{- if $.Values.ingress.tls.certManagerClusterIssuer }}
+    cert-manager.io/cluster-issuer: {{ $.Values.ingress.tls.certManagerClusterIssuer }}
+    {{- end }}
+    {{- if $.Values.ingress.annotations }}
+    {{- $.Values.ingress.annotations | toYaml | nindent 4 }}
     {{- end }}
-    kubernetes.io/ingress.class: {{ $.Values.haproxy.ingress.class }}
   name: ingress
   namespace: {{ $.Values.namespace }}
-
   labels:
     scope: jitsi
-
 spec:
+  {{ if $.Values.ingress.class -}}
+  ingressClassName: {{ $.Values.ingress.class }}
+  {{ end -}}
   rules:
-  - host: {{ $.Values.haproxy.ingress.host }}
+  {{ range $hosts -}}
+  - host: {{ . | quote }}
     http:
       paths:
+      {{ range $.Values.ingress.extraPaths -}}
+      - backend:
+          service:
+            name: {{ .serviceName | required "serviceName is required for every ingress.extraPaths" | quote }}
```

Comment:
We expect the helm user to now the name of the service names created by this helm chart? Or am I missing something?

Reply:
If the Helm user wishes to inject a specific path override into the Ingress, e.g. using the special [ALB action annotations](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#actions), then yes they need to know what they're targetting.

Reply:
Oh that's a fun feature. So current ingress path do not change, it just gives the possibility to add more ingress paths.